sdk/dotnet/Pulumi.Automation/Events/Events.cs

```diff
@@ -0,0 +1,495 @@
+// Copyright 2016-2021, Pulumi Corporation
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+// NOTE: The classes in this file are intended to align with the serialized
+// JSON types defined and versioned in sdk/go/common/apitype/events.go
+namespace Pulumi.Automation.Events
+{
+    // TODO(vipentti): Split to separate files?
+
+    /// <summary>
+    /// CancelEvent is emitted when the user initiates a cancellation of the update in progress, or
+    /// the update successfully completes.
+    /// </summary>
+    public class CancelEvent
+    {
+        internal CancelEvent() {}
+    }
+
+    /// <summary>
+    /// StdoutEngineEvent is emitted whenever a generic message is written, for example warnings
+    /// from the pulumi CLI itself. Less common than DiagnosticEvent.
+    /// </summary>
+    public class StdoutEngineEvent
+    {
+        public string Message { get; }
+
+        public string Color { get; }
+
+        internal StdoutEngineEvent(string message, string color)
+        {
+            Message = message;
+            Color = color;
+        }
+    }
+
+    /// <summary>
+    /// DiagnosticEvent is emitted whenever a diagnostic message is provided, for example errors from
+    /// a cloud resource provider while trying to create or update a resource.
+    /// </summary>
+    public class DiagnosticEvent
+    {
+        public string? Urn { get; }
+
+        public string? Prefix { get; }
+
+        public string Message { get; }
+
+        public string Color { get; }
+
+        // TODO: Make this into enum?
+        public string Severity { get; }
```

Comment:
Should Severity be its own `enum`?

Reply:
I agree that it probably could be but I haven't made it one in any of the other languages, I would leave it as-is for now.